packages/commonwealth/client/scripts/state/api/members/removeRole.ts

```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
+import { notifyError } from 'controllers/app/notifications';
+import app from 'state';
+import RoleInfo from '../../../models/RoleInfo';
+
+type RemoveRoleProps = {
+  role: RoleInfo;
+  onRoleUpdate: (oldRole: RoleInfo, newRole: RoleInfo) => void;
+};
+
+const removeRole = async ({ role, onRoleUpdate }: RemoveRoleProps) => {
+  const communityObj = { chain: app.activeChainId() };
+
+  try {
+    const res = await axios.post(`${app.serverUrl()}/upgradeMember`, {
+      ...communityObj,
+      new_role: 'member',
+      address: role.Address.address,
+      jwt: app.user.jwt,
+    });
+
+    if (res.data.status !== 'Success') {
+      throw new Error(`Got unsuccessful status: ${res.data.status}`);
+    }
+
+    const newRole = res.data.result;
+    onRoleUpdate(role, newRole);
```

Comment:
mutations should return the data from functions which can then be accessed with the `data` 
```
const { data, isLoading, ...others } = useAnyMutation()
```
lets not use callbacks for this.

Reply:
same here